### PR TITLE
feat: Profile the runtime of happy create

### LINKED ISSUE #131: As an engineer using Happy CLI, I would like to know how long do stages of the run take, so I can stay informed
Oftentimes, `happy create stack` takes a while, and due to the verbosity of the output, it is not trivial to understand how long does each phase of the run take. It will be beneficial to display a summary at the end of the run, indicating how long did the entire process take, and break it down by each key component. For example:

```
docker build: 20m
docker push: 1m
terraform: 2m
```

### PR DESCRIPTION
**Pull request by dtsai-czi**
Resolves #131

### CODE REVIEW

File: pkg/profiler/profiler.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+func (p *Profiler) AddRuntime(startTime time.Time, sectorName string) {
+	sectorDuration := time.Since(startTime)
+	p.runtimes = append(p.runtimes,
+		runtime{
+			duration:   sectorDuration,
+			sectorName: sectorName})
+}
+
+func (p *Profiler) PrintRuntimes() {
```
Comment (edulop91) [inline pkg/profiler/profiler.go:29]: There's a tableprinter you can probably use here

File: pkg/profiler/profiler.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	fmt.Println("Profiler results:")
+	for _, runtime := range p.runtimes {
+		fmt.Println("Sector", runtime.sectorName, "finished in", runtime.duration)
```
[the remark this answers is not in the document]
Comment (dtsai-czi) [reply pkg/profiler/profiler.go]: Durations already seem to automatically format to the smallest non-fractional unit of time when they're printed so it seems fine as it is--what is the general preference for human readability?

[the remark this answers is not in the document]
Comment (alexlokshin-czi) [reply pkg/profiler/profiler.go]: `fmt.Sprintf("%s", d)` will produce something like `1h5m12s`

[the remark this answers is not in the document]
Comment (alexlokshin-czi) [reply pkg/profiler/profiler.go]: Eduardo refers to `units.HumanDuration(elapsed)`

[the remark this answers is not in the document]
Comment (edulop91) [reply pkg/profiler/profiler.go]: https://go.dev/play/p/sIiPfVMOnJv -- I guess it's fine. Depending on the  duration you might get resolution that is not very helpful for humans

File: pkg/profiler/profiler.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	runtimes []runtime
+}
+
+func NewProfiler() *Profiler {
+	return &Profiler{}
+}
+
+func (p *Profiler) AddRuntime(startTime time.Time, sectorName string) {
+	sectorDuration := time.Since(startTime)
+	p.runtimes = append(p.runtimes,
```
[the remark this answers is not in the document]
Comment (dtsai-czi) [reply pkg/profiler/profiler.go:24]: From my understanding, Profiler should get initialized at the same time as ArtifactBuilder and shouldn't panic regardless of which command uses it

[the remark this answers is not in the document]
Comment (dtsai-czi) [reply pkg/profiler/profiler.go:24]: > Perhaps, a globally initialized profiler will be a better choice?

This is happening next when I attach the profiler to ctx--the basic functionality appears to be working so this seems like the next refinement of the feature

**Review by edulop91 — APPROVED** [review]

File: pkg/profiler/profiler.go (hunk 1 above)
Comment (dtsai-czi) [reply pkg/profiler/profiler.go:31]: Going to defer this to until after we get the rest of the CLI profiled

**Pull request merged**